dvc/checkout.py

```diff
@@ -310,21 +321,9 @@ def checkout(
         _remove(path_info, fs, cache, force=force)
         failed = path_info
 
-    elif not relink and not _changed(path_info, fs, obj, cache):
+    elif not relink and not _changed(path_info, fs, obj, cache, state=state):
         logger.trace("Data '%s' didn't change.", path_info)  # type: ignore
         skip = True
-    else:
-        try:
-            check(cache, obj)
```

Comment:
This check no longer required, objects passed into `checkout()` should already have been checked. In the event that a file entry within a tree does not exist for some reason, we now raise the appropriate error when linking fails